### ISSUE TITLE
NoUnused.Variables: False positive in aliased custom type

### ISSUE THREAD

**Issue by zwilias** [jfmengels/elm-review-unused#42]
**Describe the bug**
When an imported type is locally aliased with the same name, elm-review gets confused about the import being necessary and flags it as redundant, while removing it results in compilation errors.

**SSCCE (Short Self-Contained Correct Example)**

This requires a few files in order to produce the error (and only this error ;)):

### `src/A.elm`
```elm
module A exposing (T(..))


type T
    = T String
```

### `src/B.elm`
```elm
module B exposing (val)

import A


val : A.T
val =
    A.T "hi"
```

### `src/Main.elm`
```elm
module Main exposing (main)

import A exposing (T(..))
import B
import Html


type alias T =
    A.T


f : T -> String
f (T s) =
    s


main =
    Html.text (f B.val)
```

**Expected behavior**
No errors!

**Actual behavior**

```
 ❯ npx elm-review --template jfmengels/elm-review-unused/example                                                                                                                                 [09:20:03]
✔ Fetching template information
-- ELM-REVIEW ERROR ------------------------------------------ src/Main.elm:3:20

(fix) NoUnused.Variables: Imported constructors for `T` are not used

3| import A exposing (T(..))
                      ^^^^^
4| import B

You should either use this value somewhere, or remove it at the location I
pointed at.

Errors marked with (fix) can be fixed automatically using `elm-review --fix`.

I found 1 error in 1 file.
```

**Comment by jfmengels** [comment]
Thanks for the report and SSCCE! I'll try to look at it this week before holidays. Please remind me in a while if I don't get to it.

**Comment by jfmengels** [comment]
Fix available in `1.1.14`. Thanks @gampleman for fixing the issue!
